lab_2/main.py

```diff
@@ -184,23 +183,23 @@ def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
             return None
     if len(language_labels) != len(known_text_vectors):
         return None
-    # use function manhattan_distance
     if metric == 'manhattan':
-        # calculate manhattan_distance
-        distance = [calculate_distance_manhattan(unknown_text_vector, vector)
-                    for vector in known_text_vectors]
+        # calculate manhattan distances via function calculate_distance_manhattan
+        distances = [calculate_distance_manhattan(unknown_text_vector, vector)
+                     for vector in known_text_vectors]
     else:
-        distance = [calculate_distance(unknown_text_vector, vector)
-                    for vector in known_text_vectors]
-    # [1.89, 1.04, 2.24, 1.2334, 1.05, 1.6456]
+        # what if metric='euclid'? calculate euclidean distances via function calculate_distance
+        distances = [calculate_distance(unknown_text_vector, vector)
+                     for vector in known_text_vectors]
     # find knn
-    knn = sorted(zip(language_labels, distance))[:k]
-    # [('de', 1.26), ('de', 1.32), ('la', 1.53)]
+    knn = sorted(zip(language_labels, distances), key=lambda x: x[1])[:k]
```

Comment:
what is lambda and what does it do?

Reply:
Lambda is a single-line function declared with no name. It can have any number of arguments, but it can only have one expression. We need a lambda here to sort tuples (language_labels, distances) via comparing the distances (distance is an element with index "1")